spec.md

```diff
@@ -1090,6 +1119,287 @@ message ControllerServiceCapability {
+  //    an identifier by which to refer to the newly created snapshot.
+  //    If a storage system supports this, it can optionally use this
+  string name = 2;
+
+  // The description of the snapshot.
+  // This field is OPTIONAL.
+  string description = 3;
+
+  // Credentials used by Controller plugin to authenticate/authorize
+  // snapshot creation request.
+  // This field contains credential data, for example username and
+  // password. Each key must consist of alphanumeric characters, '-',
+  // '_' or '.'. Each value MUST contain a valid string. An SP MAY
+  // choose to accept binary (non-string) data by using a binary-to-text
+  // encoding scheme, like base64. An SP SHALL advertise the
+  // requirements for credentials in documentation. COs SHALL permit
+  // users to pass through the required credentials. This information is
+  // sensitive and MUST be treated as such (not logged, etc.) by the CO.
+  // This field is OPTIONAL.
+  map<string, string> controller_create_credentials = 4;
+
+  // NOTE: In the future, we may want to add “tags” to identify a
+  // snapshot. This is generated by CO internally and passed to the
+  // cloud storage backend through the plugin. This can serve the
+  // following purpose: Allow cloud storage admin to identify snapshots
+  // created by a specific CO.
+  // map<string, string> tags = 5;
+}
+
+message CreateSnapshotResponse {
+  // Contains all attributes of the newly created snapshot that are
+  // relevant to the CO along with information required by the Plugin
+  // to uniquely identify the snapshot. This field is REQUIRED.
+  Snapshot snapshot = 1;
+}
+
+// The information about a provisioned snapshot.
+message Snapshot {
+  // The restored size of the snapshot in bytes.  This indicates how
+  // much data was written on the snapshot. It’s the minimum required
+  // size when creating a volume from snapshot. This field is OPTIONAL.
+  // If not set (value of 0), it indicates this size is unknown.
+  // The value of this field MUST NOT be negative.
+  int64 restored_size = 1;
+
+  // The size of the snapshot in bytes. This indicates the given size of
+  // the snapshot when it was created. This may be equal to or greater
+  // than the restored_size. This field is OPTIONAL. If not set (value
+  // of 0), it indicates this size is unknown.
+  // The value of this field MUST NOT be negative.
+  int64 size = 2;
+
+  // Contains identity information for the created snapshot. This field
+  // is REQUIRED. The identity information will be used by the CO in
+  // subsequent calls to refer to the provisioned snapshot.
+  string id = 3;
+
+  // Description for the created snapshot. This field is OPTIONAL.
+  string description = 4;
+
+  // Identity information for the source volume. Note that create
+  // snapshot from snapshot is not supported here so the source has to
+  // be a volume. This field is REQUIRED.
+  string volume_id = 5;
```

Comment:
Just to expand on this request - The storage system is ultimately responsible for the creation of the snapshot.  There could be some delay between when the CO requests the snapshot and when the snapshot is actually taken.  In order to prevent inconsistency, a storage system driven "create time/date" would be useful.